plugins/manager/external.go

```diff
@@ -0,0 +1,37 @@
+package manager
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/nomad-autoscaler/agent/config"
+	"github.com/hashicorp/nomad-autoscaler/plugins"
+)
+
+// loadExternalPlugin takes the passed plugin and places it into the
+// PluginManager store as a record that it should be launched and dispensed.
+func (pm *PluginManager) loadExternalPlugin(cfg *config.Plugin, pluginType string) {
+
+	info := &pluginInfo{
+		args:    cfg.Args,
+		config:  cfg.Config,
+		exePath: filepath.Join(pm.pluginDir, cleanPluginExecutable(cfg.Name)),
+	}
+
+	// Add the plugin.
+	pm.pluginsLock.Lock()
+	pm.plugins[plugins.PluginID{Name: cfg.Name, PluginType: pluginType}] = info
```

Comment:
If the operator configures two plugins with the same name but they are different plugin types I don't believe there would be an overlap. Within the config they are held as separate items, partition by their type and here the map key is a composite which would allow them to coexist. If they have the same name and plugin type, then this wouldn't make it this far and be caught up in the config loading/merging, although I am unsure exactly how this would play out.

Reply:
👍